jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-client/src/main/java/org/jbpm/workbench/cm/client/roles/CaseRolesPresenter.java

```diff
@@ -48,120 +58,161 @@ public String getTittle() {
-        if(!isNullOrEmpty(userName) && !isNullOrEmpty(groupName)){
-            caseService.call(
-                    (Void) -> findCaseInstance()
-            ).assignGroupAndUserToRole(serverTemplateId, containerId, caseId, roleName, userName,groupName);
-        }else {
-            if (!isNullOrEmpty(userName)) {
-                addUserToRole(userName, roleName);
+    private CaseRoleAction getRemoveAction(String roleName, boolean enabledAction) {
+        return new CaseRoleAction() {
+            @Override
+            public String label() {
+                return translationService.format(REMOVE_ALL_ASSIGNMENTS);
             }
-            if(!isNullOrEmpty(groupName)){
-                addGroupToRole(groupName,roleName);
+
+            @Override
+            public boolean isEnabled() {
+                return enabledAction;
             }
+
+            @Override
+            public void execute() {
+                confirmPopup.show(translationService.format(REMOVE_ASSIGNMENT),
+                        translationService.format(REMOVE),
+                        translationService.format(REMOVE_ALL_USERS_GROUPS_FROM_ROLE, roleName),
+                        () -> assignToRole(roleName, Collections.emptyList(),
+                                Collections.emptyList()));
+            }
+        };
+    }
+
+    private boolean isAssignmentAvailable(CaseDefinitionSummary cds, CaseInstanceSummary cis, String roleName) {
+        final Integer roleCardinality = cds.getRoles().get(roleName);
+        if (roleCardinality == -1) {
+            return true;
         }
+        final Integer roleInstanceCardinality =
+                cis.getRoleAssignments().stream()
+                        .filter(ra -> roleName.equals(ra.getName()))
+                        .findFirst()
+                        .map(ra -> ra.getGroups().size() + ra.getUsers().size()).orElse(0);
+        return roleInstanceCardinality < roleCardinality;
     }
 
-    protected void addUserToRole(final String userName, final String roleName) {
-        caseService.call(
-                (Void) -> findCaseInstance()
-        ).assignUserToRole(serverTemplateId, containerId, caseId, roleName, userName);
+    protected CaseRoleAssignmentSummary getRoleAssignment(final CaseInstanceSummary cis, final String roleName) {
+        if (cis.getRoleAssignments() != null && cis.getRoleAssignments().size() > 0) {
+            return cis.getRoleAssignments()
+                    .stream()
+                    .filter(ra -> roleName.equals(ra.getName()))
+                    .findFirst().orElse(CaseRoleAssignmentSummary.builder().name(roleName).build());
+        }
+        return CaseRoleAssignmentSummary.builder().name(roleName).build();
+
     }
 
-    protected void addGroupToRole(final String groupName, final String roleName) {
+    protected Set<String> getRolesAvailableForAssignment(final CaseInstanceSummary cis, final CaseDefinitionSummary cds) {
```

Comment:
Where is this method used?
Where are we checking during assignments that the role cardinality hasn't been reached?
I see this method only used in Presenter test class.